.travis.yml

```
@@ -1,28 +1,102 @@
-  - OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=ON
+addons:
+  apt:
+    packages:
+      - libcppunit-dev
+      - libeigen3-dev
+      - python-future
+      - python-psutil
+      - python-sip-dev
+      - python3-future
+      - python3-psutil
+      - python3-sip-dev
+      - sip-dev
+
+matrix:
+  include:
+    - os: linux
+      dist: xenial
+      compiler: gcc
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=OFF
+    - os: linux
+      dist: xenial
+      compiler: gcc
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=ON
+    - os: linux
+      dist: xenial
+      compiler: gcc
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=OFF
+    - os: linux
+      dist: xenial
+      compiler: gcc
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=ON
+    - os: linux
+      dist: xenial
+      compiler: clang
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=OFF
+    - os: linux
+      dist: xenial
+      compiler: clang
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=ON
+    - os: linux
+      dist: xenial
+      compiler: clang
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=OFF
+    - os: linux
+      dist: xenial
+      compiler: clang
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=ON
+    - os: osx
+      osx_image: xcode8.3
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=OFF
+    - os: osx
+      osx_image: xcode8.3
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=ON
+    - os: osx
+      osx_image: xcode8.3
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=OFF
+    - os: osx
+      osx_image: xcode8.3
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=ON
+    - os: osx
+      osx_image: xcode10.1
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=OFF
+    - os: osx
+      osx_image: xcode10.1
+      env: OROCOS_KDL_BUILD_TYPE=Debug BUILD_PYKDL_PYBIND11=ON
+    - os: osx
+      osx_image: xcode10.1
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=OFF
+    - os: osx
+      osx_image: xcode10.1
+      env: OROCOS_KDL_BUILD_TYPE=Release BUILD_PYKDL_PYBIND11=ON
+
+virtualenv:
+  system_site_packages: true
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install --force cppunit eigen sip; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then pip3 install future psutil; fi
 
 before_script:
-  - sudo apt-get -qq update
-  - sudo apt-get install -y libeigen3-dev libcppunit-dev python-sip-dev python3-sip-dev python-psutil python3-psutil python-future python3-future
   # build orocos_kdl
   - cd orocos_kdl
   - mkdir build
   - cd build
   - cmake -DENABLE_TESTS:BOOL=ON -DCMAKE_CXX_FLAGS:STRING="-Wall" -DCMAKE_BUILD_TYPE=${OROCOS_KDL_BUILD_TYPE} ./..
   # compile and install orocos_kdl
-  - make
-  - sudo make install
+  - make VERBOSE=1
+  - make install
```

```suggestion
  - sudo make install
```
sudo is required in linux builds https://travis-ci.org/github/orocos/orocos_kinematics_dynamics/jobs/695656703#L646